packages/plugin-window-unhandled-rejection/src/unhandled-rejection.ts

```diff
@@ -1,82 +1,96 @@
+      const listener: Listener = (ev) => {
+        const bluebirdEvent = ev as BluebirdPromiseRejectionEvent
+
+        let error = ev.reason
         let isBluebird = false
 
         // accessing properties on evt.detail can throw errors (see #394)
         try {
-          // @ts-expect-error detail does not exist on type PromiseRejectionEvent
-          if (evt.detail && evt.detail.reason) {
-            // @ts-expect-error detail does not exist on type PromiseRejectionEvent
-            error = evt.detail.reason
+          if (bluebirdEvent.detail && bluebirdEvent.detail.reason) {
+            error = bluebirdEvent.detail.reason
             isBluebird = true
           }
         } catch (e) {}
 
-        // Report unhandled promise rejections as handled if the user has configured it
-        // @ts-expect-error _config is private API
-        const unhandled = !client._config.reportUnhandledPromiseRejectionsAsHandled
-
-        const event = client.Event.create(error, false, {
+        const event = internalClient.Event.create(error, false, {
           severity: 'error',
-          unhandled,
+          // Report unhandled promise rejections as handled when set by config
+          unhandled: !internalClient._config.reportUnhandledPromiseRejectionsAsHandled,
           severityReason: { type: 'unhandledPromiseRejection' }
-          // @ts-expect-error _logger is private API
-        }, 'unhandledrejection handler', 1, client._logger)
+        }, 'unhandledrejection handler', 1, internalClient._logger)
 
         if (isBluebird) {
           map(event.errors[0].stacktrace, fixBluebirdStacktrace(error))
         }
 
-        client._notify(event, (event) => {
+        internalClient._notify(event, (event) => {
           if (isError(event.originalError) && !event.originalError.stack) {
             event.addMetadata('unhandledRejection handler', {
               [Object.prototype.toString.call(event.originalError)]: {
                 name: event.originalError.name,
                 message: event.originalError.message,
-                // @ts-expect-error Property 'code' does not exist on type 'Error'
+                // @ts-expect-error Optional NodeJS error property
                 code: event.originalError.code
               }
             })
           }
         })
       }
-      if ('addEventListener' in win) {
+      if (typeof win.addEventListener === 'function') {
         win.addEventListener('unhandledrejection', listener)
       } else {
-        // @ts-expect-error onunhandledrejection does not exist on type never
+        // @ts-expect-error IE8 onunhandledrejection does not match the signature of the modern listener
         win.onunhandledrejection = (reason, promise) => {
-          // @ts-expect-error detail does not exist on type PromiseRejectionEvent
-          listener({ detail: { reason, promise } })
+          listener({ detail: { reason, promise } } as unknown as PromiseRejectionEvent)
         }
```

Comment:
@tomlongridge have we officially dropped support for IE8 now? this block can go if so 🔥